src/group_operations/addition.jl

```diff
@@ -0,0 +1,206 @@
+# also already handled on the `compose()` level
+Base.:+(e::Identity{AdditionGroupOperation}) = e
+Base.:+(e::Identity{AdditionGroupOperation}, ::Identity{AdditionGroupOperation}) = e
+Base.:+(::Identity{AdditionGroupOperation}, g) = g
+Base.:+(g, ::Identity{AdditionGroupOperation}) = g
+
+Base.:-(e::Identity{AdditionGroupOperation}) = e
+Base.:-(e::Identity{AdditionGroupOperation}, ::Identity{AdditionGroupOperation}) = e
+Base.:-(::Identity{AdditionGroupOperation}, g) = -g
+Base.:-(g, ::Identity{AdditionGroupOperation}) = g
+
+_doc_compose_add = """
+    compose(G::LieGroup{𝔽,AdditionGroupOperation}, g, h)
+    compose!(G::LieGroup{𝔽,AdditionGroupOperation}, k, g, h)
+
+Copmute the group operation composition of `g` and `h` with respect to
+the [`AdditionGroupOperation`](@ref) on `G`, which falls back to calling
+`g+h`, where `+` is assumed to be overloaded accordingly.
+
+This can be computed in-place of `k`.
+"""
+
+@doc "$(_doc_compose_add)"
+compose(::LieGroup{𝔽,AdditionGroupOperation}, g, h) where {𝔽}
+
+@doc "$(_doc_compose_add)"
+compose!(::LieGroup{𝔽,AdditionGroupOperation}, k, g, h) where {𝔽}
+
+function _compose!(G::LieGroup{𝔽,AdditionGroupOperation}, k, g, h) where {𝔽}
+    ManifoldsBase.copyto!(G, k, g + h)
+    return k
+end
+
+_doc_diff_conjugate_add = """
+    diff_conjugate((G::LieGroup{𝔽,AdditionGroupOperation}, Y, g, h, X))
+    diff_conjugate!(G::LieGroup{𝔽,AdditionGroupOperation}, Y, g, h, X)
+
+Compute the differential of the conjutage ``c_g(h) = g$(_math(:∘))h$(_math(:∘))g^{-1} = g+h-g = h``,
+which simplifies for [`AdditionGroupOperation`](@ref) to ``D(c_g(h))[X] = X``.
+"""
+
+@doc "$(_doc_diff_conjugate_add)"
+diff_conjugate(G::LieGroup{𝔽,AdditionGroupOperation}, g, h, X) where {𝔽}
+
+@doc "$(_doc_diff_conjugate_add)"
+function diff_conjugate!(G::LieGroup{𝔽,AdditionGroupOperation}, Y, g, h, X) where {𝔽}
+    return copyto!(LieAlgebra(G), Y, X)
+end
+
+_doc_diff_inv_add = """
+    diff_inv(G::LieGroup{𝔽,AdditionGroupOperation}, g, X)
+    diff_inv!(G::LieGroup{𝔽,AdditionGroupOperation}, Y, g, X)
+
+Compute the differential of the inverse operation ``ι_{$(_math(:G))}(g) = g^-1 = -g``,
+which simplifies for [`AdditionGroupOperation`](@ref) to ``Dι_{$(_math(:G))}(g)[X] = -X``
+"""
+
+@doc "$(_doc_diff_inv_add)"
+diff_inv(G::LieGroup{𝔽,AdditionGroupOperation}, g, X) where {𝔽}
+
+@doc "$(_doc_diff_inv_add)"
+function diff_inv!(G::LieGroup{𝔽,AdditionGroupOperation}, Y, g, X) where {𝔽}
+    return copyto!(LieAlgebra(G), Y, -X)
+end
+
+_doc_diff_left_compose_add = """
+    diff_left_compose(G::LieGroup{𝔽,AdditionGroupOperation}, g, h, X)
+    diff_left_compose!(G::LieGroup{𝔽,AdditionGroupOperation}, Y, g, h, X)
+
+Compute the differential of the left group multiplication ``λ_g(h) = g$(_math(:∘))h``,
+which simplifies for [`AdditionGroupOperation`](@ref) to ``Dλ_g(h)[X] = X``.
+"""
+
+@doc "$(_doc_diff_left_compose_add)"
+diff_left_compose(G::LieGroup{𝔽,AdditionGroupOperation}, g, h, X) where {𝔽}
```

Comment:
```suggestion
diff_left_compose(G::LieGroup{𝔽,AdditionGroupOperation}, g, h, X) where {𝔽} = X
```
IMO it makes sense to have it implemented like this to avoid allocations in the fallback.

Reply:
see above, most other allocating variants allocate, then this one does not, which might be confusing if you expect them to.